source/common/router/router.cc

```diff
@@ -576,10 +576,21 @@ void Filter::handleNon5xxResponseHeaders(const Http::HeaderMap& headers, bool en
   }
 }
 
+void Filter::onUpstream100ContinueHeaders(Http::HeaderMapPtr&& headers) {
+  ENVOY_STREAM_LOG(debug, "upstream 100 continue", *callbacks_);
+
+  downstream_response_started_ = true;
+  // We will double count response codes for 100-Continue.
+  upstream_request_->upstream_host_->outlierDetector().putHttpResponseCode(100);
```

Comment:
nit: use constant in codes.h, also can you verify the outlier detector code isn't going to do something stupid with a 100 status code and maybe add some tests there? (Not even sure if we should send 100 to the outlier detector, I guess that is worth discussing).

Reply:
Now that I think of it, given 100-Continue is timing based (clients can choose to post data right away, then servers nto send it) and likely rare, I'm inclined to not send it.